### PR TITLE
Fix a bunch of warnings

### PR DESCRIPTION
**Pull request by swills**
Pull request checklist:

- [X] `CHANGELOG.md` was updated, if applicable

# Description

Fixes a bunch of warnings

### CODE REVIEW

File: src/RunGui.cpp [hunk 1]
```diff
@@ -111,9 +111,12 @@ namespace {
         QFile::remove(path);
     }
 
+#ifndef C_DEBUG
```
Comment (fourtf) [inline src/RunGui.cpp:114]: how is this related to warning?

File: src/common/DownloadManager.hpp [hunk 2]
```diff
@@ -21,7 +21,7 @@ class DownloadManager : public QObject
 
 private:
     QNetworkAccessManager *manager;
-    QNetworkReply *reply;
+    QNetworkReply *downloadReply;
```
Comment (fourtf) [inline src/common/DownloadManager.hpp:24]: does this fix a warning?

Comment (fourtf) [reply src/common/DownloadManager.hpp:24]: and if so, which one

File: src/controllers/accounts/AccountModel.cpp [hunk 3]
```diff
@@ -26,7 +26,7 @@ void AccountModel::getRowFromItem(const std::shared_ptr<Account> &item,
 }
 
 int AccountModel::beforeInsert(const std::shared_ptr<Account> &item,
-                               std::vector<QStandardItem *> &row,
+                               std::vector<QStandardItem *> &,
```
Comment (fourtf) [inline src/controllers/accounts/AccountModel.cpp:29]: The argument missing is way more confusing here. Try changing it to `std::vector<QStandardItem *> &, /* row */,`

File: src/controllers/commands/CommandController.cpp [hunk 4]
```diff
@@ -228,15 +228,15 @@ void CommandController::initialize(Settings &, Paths &paths)
         this->items_.append(command);
     }
 
-    this->registerCommand("/debug-args", [](const auto &words, auto channel) {
+    this->registerCommand("/debug-args", [](const auto &, auto channel) {
```
Comment (fourtf) [inline src/controllers/commands/CommandController.cpp:231]: same here as above

File: src/messages/layouts/MessageLayout.cpp [hunk 5]
```diff
@@ -63,7 +63,7 @@ int MessageLayout::getHeight() const
 
 // Layout
 // return true if redraw is required
-bool MessageLayout::layout(int width, float scale, MessageElementFlags flags)
+bool MessageLayout::layout(int width, float scale, MessageElementFlags msgflags)
```
Comment (fourtf) [inline src/messages/layouts/MessageLayout.cpp:66]: Why was this renamed?
Apart from that it breaks the `camelCase` naming convention by being all lowercase.

File: src/messages/layouts/MessageLayout.cpp [hunk 6]
```diff
@@ -112,13 +112,13 @@ bool MessageLayout::layout(int width, float scale, MessageElementFlags flags)
     return true;
 }
 
-void MessageLayout::actuallyLayout(int width, MessageElementFlags flags)
+void MessageLayout::actuallyLayout(int width, MessageElementFlags mflags)
```
Comment (fourtf) [inline src/messages/layouts/MessageLayout.cpp:115]: Same as above but here it's a different name for no reason?!

File: src/messages/layouts/MessageLayoutContainer.cpp [hunk 7]
```diff
@@ -378,13 +378,13 @@ void MessageLayoutContainer::paintSelection(QPainter &painter, int messageIndex,
                 continue;
             }
 
-            for (int i = line.startIndex; i < line.endIndex; i++)
```
Comment (fourtf) [inline src/messages/layouts/MessageLayoutContainer.cpp:381]: why was this renamed to `ii`?

File: src/messages/layouts/MessageLayoutContainer.cpp [hunk 8]
```diff
@@ -394,18 +394,18 @@ void MessageLayoutContainer::paintSelection(QPainter &painter, int messageIndex,
                     {
                         returnAfter = true;
                         index = line.startCharIndex;
-                        for (int i = line.startIndex; i < line.endIndex; i++)
```
Comment (fourtf) [inline src/messages/layouts/MessageLayoutContainer.cpp:397]: why is this `kk` instead of something descriptive?

File: src/messages/layouts/MessageLayoutContainer.cpp [hunk 9]
```diff
@@ -378,13 +378,13 @@ void MessageLayoutContainer::paintSelection(QPainter &painter, int messageIndex,
                 continue;
             }
 
-            for (int i = line.startIndex; i < line.endIndex; i++)
+            for (int ii = line.startIndex; ii < line.endIndex; ii++)
             {
-                int c = this->elements_[i]->getSelectionIndexCount();
```
Comment (fourtf) [inline src/messages/layouts/MessageLayoutContainer.cpp:383]: why was this changed at all?

File: src/providers/chatterino/ChatterinoBadges.cpp [hunk 10]
```diff
@@ -13,7 +13,7 @@
 #include <map>
 
 namespace chatterino {
-void ChatterinoBadges::initialize(Settings &settings, Paths &paths)
+void ChatterinoBadges::initialize(Settings &, Paths &)
```
Comment (fourtf) [inline src/providers/chatterino/ChatterinoBadges.cpp:16]: Here it's obvious what the variables are so I'm fine with this. In some other cases where there's only a `bool` you have no idea what it is without knowing the name of the variable.

File: src/providers/irc/AbstractIrcServer.cpp [hunk 11]
```diff
@@ -267,14 +267,14 @@ ChannelPtr AbstractIrcServer::getChannelOrEmpty(const QString &dirtyChannelName)
 std::vector<std::weak_ptr<Channel>> AbstractIrcServer::getChannels()
 {
     std::lock_guard lock(this->channelMutex);
-    std::vector<std::weak_ptr<Channel>> channels;
+    std::vector<std::weak_ptr<Channel>> rchannels;
```
Comment (fourtf) [inline src/providers/irc/AbstractIrcServer.cpp:270]: Rename to something that doesn't have a pointless prefix like `result`

File: src/providers/irc/AbstractIrcServer.hpp [hunk 12]
```diff
@@ -62,7 +62,7 @@ class AbstractIrcServer : public QObject
         const QString &channelName) = 0;
 
     virtual void privateMessageReceived(Communi::IrcPrivateMessage *message);
-    virtual void readConnectionMessageReceived(Communi::IrcMessage *message);
+    virtual void readConnectionMessageReceived(Communi::IrcMessage *);
```
Comment (fourtf) [inline src/providers/irc/AbstractIrcServer.hpp:65]: why is this changed in the header?

File: src/providers/twitch/api/Helix.hpp [hunk 13]
```diff
@@ -189,7 +189,7 @@ class Helix final : boost::noncopyable
 private:
     NetworkRequest makeRequest(QString url, QUrlQuery urlQuery);
 
-    QString clientId;
+    QString hclientId;
```
Comment (fourtf) [inline src/providers/twitch/api/Helix.hpp:192]: random prefix that doesn't carry any meaning and is simply confusing

File: src/providers/twitch/api/Kraken.hpp [hunk 14]
```diff
@@ -53,8 +53,8 @@ class Kraken final : boost::noncopyable
 private:
     NetworkRequest makeRequest(QString url, QUrlQuery urlQuery);
 
-    QString clientId;
-    QString oauthToken;
+    QString kclientId;
+    QString koauthToken;
```
Comment (fourtf) [inline src/providers/twitch/api/Kraken.hpp:57]: same as above but it's a different random prefix?!

File: src/widgets/settingspages/SettingsPage.cpp [hunk 15]
```diff
@@ -21,36 +20,36 @@ bool filterItemsRec(QObject *object, const QString &query)
             widget->update();
         };
 
-        if (auto x = dynamic_cast<SCheckBox *>(child); x)
+        if (auto xx = dynamic_cast<SCheckBox *>(child); xx)
```
Comment (fourtf) [inline src/widgets/settingspages/SettingsPage.cpp:23]: weird variable names again

**Comment by fourtf** [comment]
This PR seems to be tunnel-focused on trying to remove as many warnings without considering how the readability of the code changes. In many scenarios I would rather see warnings like "unused variable" to be disabled rather then hiding the names.

File: src/RunGui.cpp (hunk 1 above)
Comment (swills) [reply src/RunGui.cpp:114]: In master, the warning is:

`RunGui.cpp:117:23: warning: unused function 'handleSignal'`

Because the function use is behind `#ifndef C_DEBUG`, so I put the function definition behind `#ifndef C_DEBUG` too, which leads to the variable not being used:

`RunGui.cpp:114:43: warning: unused variable 'signalsInitTime'`

File: src/common/DownloadManager.hpp (hunk 2 above)
Comment (swills) [reply src/common/DownloadManager.hpp:24]: As it says in the commit message:

`warning C4458: declaration of 'reply' hides class member`

This is on the Windows build.

File: src/messages/layouts/MessageLayout.cpp (hunk 5 above)
Comment (swills) [reply src/messages/layouts/MessageLayout.cpp:66]: As the commit message says:

`warning C4458: declaration of 'flags' hides class member`

I'm happy to rename it. What should it be? msgFlags?

File: src/messages/layouts/MessageLayout.cpp (hunk 6 above)
Comment (swills) [reply src/messages/layouts/MessageLayout.cpp:115]: Sorry for the inconsistency, I'm happy to rename it.

File: src/messages/layouts/MessageLayoutContainer.cpp (hunk 7 above)
Comment (swills) [reply src/messages/layouts/MessageLayoutContainer.cpp:381]: The warning was:

```
    warning C4456: declaration of 'i' hides previous local declaration
    warning C4456: declaration of 'c' hides previous local declaration
```

File: src/messages/layouts/MessageLayoutContainer.cpp (hunk 8 above)
Comment (swills) [reply src/messages/layouts/MessageLayoutContainer.cpp:397]: Happy to rename it, but it did seem confusing to use `i` again inside a loop that is already using `i`.

File: src/messages/layouts/MessageLayoutContainer.cpp (hunk 9 above)
Comment (swills) [reply src/messages/layouts/MessageLayoutContainer.cpp:383]: The warning was:

`warning C4456: declaration of 'c' hides previous local declaration`

File: src/providers/irc/AbstractIrcServer.cpp (hunk 11 above)
Comment (swills) [reply src/providers/irc/AbstractIrcServer.cpp:270]: FWIW, the warning was:

`warning C4458: declaration of 'channels' hides class member`

Comment (swills) [reply src/providers/irc/AbstractIrcServer.cpp:270]: Do you have a preference on name here?

File: src/providers/irc/AbstractIrcServer.hpp (hunk 12 above)
Comment (swills) [reply src/providers/irc/AbstractIrcServer.hpp:65]: The source warning was:

`warning: unused parameter 'message'`

After removing it there, I thought I saw a warning about the header. I don't see that now. Maybe I'm missing something.

File: src/providers/twitch/api/Helix.hpp (hunk 13 above)
Comment (swills) [reply src/providers/twitch/api/Helix.hpp:192]: What should it be called?

File: src/providers/twitch/api/Kraken.hpp (hunk 14 above)
Comment (swills) [reply src/providers/twitch/api/Kraken.hpp:57]: Happy to rename, what would you prefer?

File: src/widgets/settingspages/SettingsPage.cpp (hunk 15 above)
Comment (swills) [reply src/widgets/settingspages/SettingsPage.cpp:23]: Do you have a preference on name?

Comment (swills) [reply src/widgets/settingspages/SettingsPage.cpp:23]: (FWIW, I tend to use variable names that are at least 2 letters so that they're easier to search for)

File: src/controllers/accounts/AccountModel.cpp (hunk 3 above)
Comment (swills) [reply src/controllers/accounts/AccountModel.cpp]: Done

File: src/controllers/commands/CommandController.cpp (hunk 4 above)
Comment (swills) [reply src/controllers/commands/CommandController.cpp]: done

**Comment by swills** [comment]
> This PR seems to be tunnel-focused on trying to remove as many warnings without considering how the readability of the code changes. In many scenarios I would rather see warnings like "unused variable" to be disabled rather then hiding the names.

I guess to me it seems easier and better to add comments than disable the warnings, but I guess it is a matter of opinion. If you're not interested in these changes that's fine, close the PR if you like.

File: src/providers/twitch/api/Kraken.hpp (hunk 14 above)
Comment (jammehcow) [reply src/providers/twitch/api/Kraken.hpp:57]: Maybe `apiClientId` or `krakenClientId`. Probably the former since we're in the `Kraken` class

File: src/widgets/settingspages/SettingsPage.cpp (hunk 15 above)
Comment (jammehcow) [reply src/widgets/settingspages/SettingsPage.cpp:23]: `checkBox`, `label`, `comboBox`?

**Comment by ALazyMeme** [comment]
pr is stale
